tldraw/tldraw
# Support auto switching wheel behavior

This implements support for setting wheelBehavior to auto. What this means is that it will try to detect if you're using a mouse or a touchpad, and use wheelBehavior zoom when a mouse is used and pan when a touchpad is used. This is more intuitive behavior than always having it set to wheel or pan in my opinion.

As far as I know, it's unfortunately not possible to know if wheel events are generated from a mouse or a touchpad. The heuristic implemented is that if you scroll horizontally it assumes you use a touchpad and switch to pan, and if you pan with the middle mouse button it assumes you're using a mouse and switches to zoom. This seems to be similar to what other similar software does.

Fixes #4579

### Change type


### Test plan

1. Use a touchpad to pan the board by scrolling
2. Switch to a mouse and pan the board by dragging with the middle mouse button
3. Scroll and observe that it now zooms on scroll
4. Switch back to the touchpad, scroll horizontally and observe that it pans again


### Release notes

- Support automatically switching wheelBehavior when switching between touchpad and mouse (as determined by a heuristic). Use `pan` when using a touchpad and `zoom` when using a mouse.
